src/main/kotlin/com/soywiz/korge/intellij/util/libraries_utils.kt

```diff
@@ -0,0 +1,197 @@
+
+    println("Iterating through all libraries:")
+    val gradleLibraries = libraryTable.libraries.filter { library ->
+        val libraryName = library.name ?: return@filter false
+        if (!libraryName.startsWith("Gradle: ")) return@filter false
+        true
+    }
+
+    gradleLibraries.forEach { library ->
+        val libraryName = library.name!!
+        println("Parsing library: $libraryName")
+        val gradleParsed = libraryName.removePrefix("Gradle: ").split(":")
+        val parsedLibraryName = when (gradleParsed.size) {
+            2 -> {
+                ParsedLibraryName.Double(
+                    gradleParsed[0],
+                    gradleParsed[1],
+                )
+            }
+
+            3 -> {
+                ParsedLibraryName.Triple(
+                    gradleParsed[0],
+                    gradleParsed[1],
+                    gradleParsed[2],
+                )
+            }
+
+            4 -> {
+                ParsedLibraryName.Quad(
+                    gradleParsed[0],
+                    gradleParsed[1],
+                    gradleParsed[2],
+                    gradleParsed[3],
+                )
+            }
+
+            5 -> {
+                ParsedLibraryName.Penta(
+                    gradleParsed[0],
+                    gradleParsed[1],
+                    gradleParsed[2],
+                    gradleParsed[3],
+                    gradleParsed[4],
+                )
+            }
+
+            else -> {
+                println("Unsupported name: $gradleParsed")
+                return@forEach
+            }
+        }
+
+        val parsedClassRoots = library.getUrls(OrderRootType.CLASSES).map {
+            ParsedClassRoot(it!!)
+        }
+        val parsedSourceRoots = library.getUrls(OrderRootType.SOURCES).map {
+            ParsedSourceRoot(it!!)
+        }
+
+        parsedLibraries.add(
+            ParsedLibrary(
+                parsedLibraryName,
+                parsedClassRoots,
+                parsedSourceRoots,
+                library
+            )
+        )
+
+        //            println(library.rootProvider.getUrls(OrderRootType.CLASSES).toList())
+        //            println(library.rootProvider.getUrls(OrderRootType.SOURCES).toList())
+        //            println(library.getFiles(OrderRootType.CLASSES).toList())
+        //            println()
+    }
+
+    //        println("Parsed libraries:")
+    //        println(parsedLibaries.joinToString("\n"))
+    //        println()
+
+    val korgeLibraries = parsedLibraries.filter { it.name.isKorgeLibrary() }
+
+    val korgeJvmLibraries = korgeLibraries.filter {
+        it.name.isKorgeJvmLibrary()
+    }
+
+//    println("Korge JVM libraries")
+//    println(korgeJvmLibraries.joinToString("\n"))
+
+    val groupIdToJvmLibrary = korgeJvmLibraries.associateBy {
+        it.name.groupId
+    }
+
+    println("Adding JVM source to applicable Korge libraries.")
```

Comment:
Shouldn't we add also to the common, and other targets? For example, there is some code only available to some targets that are not JVM.